src/ducks/onboarding/EnsureHasAccounts.jsx

```diff
@@ -27,41 +33,75 @@ const LayoutContent = props => (
 /**
  * Replaces its children by Onboarding if we have no accounts
  */
-const EnsureHasAccounts = props => {
-  const { children, accounts, location } = props
-
-  if (isCollectionLoading(accounts)) {
-    return (
-      <LayoutContent>
-        <div className={styles.Onboarding__loading}>
-          <Loading />
-        </div>
-      </LayoutContent>
-    )
+class EnsureHasAccounts extends Component {
+  state = {
+    intervalId: false
+  }
+
+  startInterval = () => {
+    const { accounts, fetchAccounts } = this.props
+    if (accounts.length === 0 && !this.state.intervalId) {
+      const intervalId = setInterval(() => fetchAccounts(), 30000)
+      this.setState({ intervalId })
+    }
+  }
+
+  stopInterval = () => {
+    const { intervalId } = this.state
+    if (intervalId) {
+      clearInterval(intervalId)
+      this.setState({ intervalId: false })
+    }
+  }
+
+  componentDidMount() {
+    this.startInterval()
   }
 
-  if (
-    (accounts.data && accounts.data.length === 0) ||
-    hasParameter(location.query, 'onboarding')
-  ) {
-    return (
-      <LayoutContent>
-        <Onboarding />
-      </LayoutContent>
-    )
+  componentWillUnmount() {
+    this.stopInterval()
   }
 
-  return children
+  render() {
+    const { children, accounts, accountsCollection, location } = this.props
+
+    if (isCollectionLoading(accountsCollection)) {
+      return (
+        <LayoutContent>
+          <div className={styles.Onboarding__loading}>
+            <Loading />
+          </div>
+        </LayoutContent>
+      )
+    }
+
+    if (
+      (accounts && accounts.length === 0) ||
+      hasParameter(location.query, 'onboarding')
+    ) {
+      return (
+        <LayoutContent>
+          <Onboarding />
+        </LayoutContent>
+      )
+    }
+
+    this.stopInterval()
```

Comment:
I find it weird to execute this function in the `render` method.

I think doing the following would be better : 

* `componentDidMount`: if there are no accounts, `startInterval`
* `componentDidUpdate`: if there are some accounts, `stopInterval`
* and store the interval id outside of the state, since it's not used in the render method (put it in `this.intervalID`, for example)